malaria-iosTests/TestMedicineStockManager.swift

```diff
@@ -0,0 +1,134 @@
+import UIKit
+import XCTest
+@testable import malaria_ios
+
+class TestMedicineStockManager: XCTestCase {
+  
+  let secondsTillYesterday: Double = -86400.0
+  let twoDaysAgo: Double = -86400.0 * 2
+  let secondTillTomorrow: Double = 86400.0
+  
+  var m: MedicineManager!
+  var msm: MedicineStockManager!
+  
+  let currentPill = Medicine.Pill.Malarone
+  
+  var currentContext: NSManagedObjectContext!
+  
+  override func setUp() {
+    super.setUp()
+    
+    currentContext = CoreDataHelper.sharedInstance.createBackgroundContext()
+    m = MedicineManager(context: currentContext)
+    m.registerNewMedicine(currentPill.name(), interval: currentPill.interval())
+    m.setCurrentPill(currentPill.name())
+  
+    msm = MedicineStockManager(medicine: m.getCurrentMedicine()!)
+  }
+  
+  override func tearDown() {
+    super.tearDown()
+    m.clearCoreData()
+  }
+  
+  // Checks if Core Date successfully updates remaining pills for medicine
+  func testUpdateStock() {
+    guard let medicine = m.getCurrentMedicine() else {
+      XCTFail("Fail initializing:")
+      return
+    }
+    
+    medicine.remainingMedicine = 1
+    
+    msm.updateStock(true)
+    XCTAssertEqual(medicine.remainingMedicine, 0)
+
+    msm.updateStock(false)
+    XCTAssertEqual(medicine.remainingMedicine, 1)
+  }
+  
+  func testUserHasEnoughPillsToPressYes() {
```

Comment:
Add tests when tookMedicine is set to false. As expected, the medicineStock should remain.